### ISSUE TITLE
Enhancement - let choices be a dictionary of {key:value}, shows value but answer is key

### ISSUE THREAD

**Issue by WilliamStam** [tmbo/questionary#94]
would be cool if this could work like a `<select>` box. having an id and a label type thing. right now it just returns the selected text and using this trick https://github.com/tmbo/questionary/issues/53 you could get what index you need but that now means setting up 2 lists, 1 with the label and another with the id of the selected choice and then going through the motions of getting the index of the choice and then finding the id from the id list using the index

while this IS possible i think a lot of use cases might be improved if it had an id and label. maybe pass it a choices dictionary 
````
"What toppings do you want?"
{
    "cheese": "Super creamy nice cheese",
    "meat": "our mystry meat package"
    "vegetables": "only the freshest veggies"
}
````

shows as:
````
What toppings do you want?
> Super creamy nice cheese
   our mystry meat package
   only the freshest veggies
````
and answer is 
````
"cheese"
````

this could be a rather cool enhancement

--------
potentially add a flag to use the key as the shortcut key as well would be amazing. 

-------
basically being able to specify our own "shortcut" values instead of 0 - 9 then a - z etc.

**Comment by WilliamStam** [comment]
(sorry i pressed enter by mistake)

**Comment by kiancross** [comment]
@WilliamStam I think you can already do this:

```python3
import questionary

options = { 
    "cheese": "Super creamy nice cheese",
    "meat": "our mystry meat package",
    "vegetables": "only the freshest veggies",
}

answer = questionary.select(
    "What toppings do you want?",
    choices=[
        questionary.Choice(title=title, value=value) for value, title in options.items()
    ],  
).ask()

print(answer)
```

**Comment by WilliamStam** [comment]
wow. ok thanks! i seriously tried a ton of things to get it to work. eventualy settled on a hack of 

````
instances = list(map(lambda x: str(x.split("|")[0]).strip(), questionary.checkbox(
                "Select instance(s)",
                choices=[f"{str(x.id).ljust(3,' ')} | {str(truncate_text(x.name,chars=20)).ljust(20,' ')}" for x in instance_list]
            ).unsafe_ask()))
````

which pretty much makes the "title" `3 | name of instance` then strip it based on the pipe char to get the "id" of the value. 

this changes EVERYTHING! thank you!
